plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.test.ts

```diff
@@ -65,122 +70,147 @@ describe('fetch:cookiecutter', () => {
 
   const action = createFetchCookiecutterAction({
     integrations,
-    templaters,
+    containerRunner,
     reader: mockReader,
   });
 
-  templaters.register('cookiecutter', cookiecutterTemplater);
-
   beforeEach(() => {
-    mock({ [`${mockContext.workspacePath}/result`]: {} });
-    jest.restoreAllMocks();
+    jest.resetAllMocks();
+
+    mockContext = {
+      input: {
+        url: 'https://google.com/cookie/cutter',
+        targetPath: 'something',
+        values: {
+          help: 'me',
+        },
+      },
+      baseUrl: 'somebase',
+      workspacePath: mockTmpDir,
+      logger: getVoidLogger(),
+      logStream: new PassThrough(),
+      output: jest.fn(),
+      createTemporaryDirectory: jest.fn().mockResolvedValue(mockTmpDir),
+    };
+
+    // mock the temp directory
+    mock({ [mockTmpDir]: {} });
+    mock({ [`${join(mockTmpDir, 'template')}`]: {} });
+
+    commandExists.mockResolvedValue(null);
+
+    // Mock when run container is called it creates some new files in the mock filesystem
+    containerRunner.runContainer.mockImplementation(async () => {
+      mock({
+        [`${join(mockTmpDir, 'intermediate')}`]: {
+          'testfile.json': '{}',
+        },
+      });
+    });
+
+    // Mock when runCommand is called it creats some new files in the mock filesystem
+    runCommand.mockImplementation(async () => {
+      mock({
+        [`${join(mockTmpDir, 'intermediate')}`]: {
+          'testfile.json': '{}',
+        },
+      });
+    });
   });
 
   afterEach(() => {
     mock.restore();
   });
 
-  it('should call fetchContents with the correct values', async () => {
-    await action.handler(mockContext);
-
-    expect(fetchContents).toHaveBeenCalledWith({
-      reader: mockReader,
-      integrations,
-      baseUrl: mockContext.baseUrl,
-      fetchUrl: mockContext.input.url,
-      outputPath: resolvePath(
-        mockContext.workspacePath,
-        `template/{{cookiecutter and 'contents'}}`,
-      ),
-    });
-  });
+  it('should throw an error when copyWithoutRender is not an array', async () => {
+    (mockContext.input as any).copyWithoutRender = 'not an array';
 
-  it('should execute the cookiecutter templater with the correct values', async () => {
-    await action.handler(mockContext);
-
-    expect(cookiecutterTemplater.run).toHaveBeenCalledWith({
-      workspacePath: mockTmpDir,
-      logStream: mockContext.logStream,
-      values: mockContext.input.values,
-    });
+    await expect(
+      action.handler(mockContext),
+    ).rejects.toThrowErrorMatchingInlineSnapshot(
```

Comment:
why not just `toThrow`? 🤔

Reply:
It was just a pattern I saw that's pretty nice for when you change error messages it makes updating the tests easier because they're snapshots. :shrug: Can change.

Reply:
Nah just curious